src/utils.ts

```diff
@@ -88,6 +89,83 @@ export function generateName(from: string, usedNames: Set<string>) {
   return name
 }
 
+export interface CustomNameFunctionOptions {
+  /**
+   * the schema node which has to be named with things like `id`, `title`, `description`
+   */
+  schema: JSONSchema
+  /**
+   * this is the key under which this schema node was found, either value from properties or defintions
+   */
+  keyName?: string
+  /**
+   * this is the key of this schema object under definitions if applicable, this may be the same as `keyName`
+   */
+  definitionKeyName?: string
+  /**
+   * this is a set of all names that have already been used
+   */
+  usedNames: Set<string>
+}
```

Comment:
No need for named arguments. Just make `customName` take 4 arguments. Let's also make sure that `customName` returns a `string` (rather than `string | undefined`).